### PR TITLE
Fix EZP-24955: Dynamically inject limitation map into RoleService

### PR DESCRIPTION
**Pull request by lolautruche**
> https://jira.ez.no/browse/EZP-24955

This PR just injects policies that are being provided by policy providers into the Repository.
Tests have been updated accordingly.

### CODE REVIEW

**Comment by lolautruche** [comment]
Review ping @andrerom @bdunogier @pspanja @glye

File: eZ/Bundle/EzPublishCoreBundle/ApiLoader/RepositoryFactory.php [hunk 1]
```diff
@@ -46,14 +46,21 @@ class RepositoryFactory extends ContainerAware
...
         ConfigResolverInterface $configResolver,
         FieldTypeCollectionFactory $fieldTypeCollectionFactory,
-        $repositoryClass
+        $repositoryClass,
+        array $limitationMap
     ) {
         $this->configResolver = $configResolver;
         $this->fieldTypeCollectionFactory = $fieldTypeCollectionFactory;
         $this->repositoryClass = $repositoryClass;
+        $this->limitationMap = $limitationMap;
```
Comment (andrerom) [inline eZ/Bundle/EzPublishCoreBundle/ApiLoader/RepositoryFactory.php]: assuming this is what you refer to as `policyMap`, feel free to rename variable here, in Repository and Role Service if you want to keep it consistent.

**Comment by andrerom** [comment]
Besides comment, +1

File: eZ/Bundle/EzPublishCoreBundle/ApiLoader/RepositoryFactory.php (hunk 1 above)
Comment (lolautruche) [inline eZ/Bundle/EzPublishCoreBundle/ApiLoader/RepositoryFactory.php]: Fixed, thanks.

**Comment by glye** [comment]
+1, I guess.

**Pull request merged**